libsolidity/analysis/ContractLevelChecker.cpp

```diff
@@ -156,55 +156,53 @@ void ContractLevelChecker::findDuplicateDefinitions(map<string, vector<T>> const
-	// flag to indicate whether it is fully implemented.
-	using FunTypeAndFlag = std::pair<FunctionTypePointer, bool>;
-	map<string, vector<FunTypeAndFlag>> functions;
+	// Collects functions, static variable getters and modifiers. If they
+	// override (unimplemented) base class ones, they are replaced.
+	set<OverrideProxy, OverrideProxy::CompareBySignature> proxies;
 
-	auto registerFunction = [&](Declaration const& _declaration, FunctionTypePointer const& _type, bool _implemented)
+	auto registerProxy = [&proxies](OverrideProxy const& _overrideProxy)
 	{
-		auto& overloads = functions[_declaration.name()];
-		auto it = find_if(overloads.begin(), overloads.end(), [&](FunTypeAndFlag const& _funAndFlag)
+		auto it = proxies.find(_overrideProxy);
+		if (it != proxies.end())
 		{
-			return _type->hasEqualParameterTypes(*_funAndFlag.first);
-		});
-		if (it == overloads.end())
-			overloads.emplace_back(_type, _implemented);
-		else if (_implemented)
-			it->second = true;
+			if (!_overrideProxy.unimplemented())
+			{
+				auto _it = std::next(it);
+				proxies.erase(it);
+				proxies.insert(_it, _overrideProxy);
+			}
+		}
+		else
+			proxies.insert(_overrideProxy);
 	};
 
-	// Search from base to derived, collect all functions and update
-	// the 'implemented' flag.
+	// Search from base to derived, collect all functions and modifiers and
+	// update proxies.
 	for (ContractDefinition const* contract: boost::adaptors::reverse(_contract.annotation().linearizedBaseContracts))
 	{
 		for (VariableDeclaration const* v: contract->stateVariables())
 			if (v->isPartOfExternalInterface())
-				registerFunction(*v, TypeProvider::function(*v), true);
+				registerProxy(OverrideProxy(v));
 
 		for (FunctionDefinition const* function: contract->definedFunctions())
 			if (!function->isConstructor())
-				registerFunction(
-					*function,
-					TypeProvider::function(*function)->asCallableFunction(false),
-					function->isImplemented()
-				);
+				registerProxy(OverrideProxy(function));
+
+		for (ModifierDefinition const* modifier: contract->functionModifiers())
+			registerProxy(OverrideProxy(modifier));
 	}
 
 	// Set to not fully implemented if at least one flag is false.
-	// Note that `_contract.annotation().unimplementedFunctions` has already been
+	// Note that `_contract.annotation().unimplementedDeclarations` has already been
 	// pre-filled by `checkBaseConstructorArguments`.
-	for (auto const& it: functions)
-		for (auto const& funAndFlag: it.second)
-			if (!funAndFlag.second)
-			{
-				FunctionDefinition const* function = dynamic_cast<FunctionDefinition const*>(&funAndFlag.first->declaration());
-				solAssert(function, "");
-				_contract.annotation().unimplementedFunctions.push_back(function);
-				break;
-			}
+	for (auto const& proxy: proxies)
+		if (proxy.unimplemented())
+		{
+			_contract.annotation().unimplementedDeclarations.push_back(proxy.declaration());
+			break;
```

Comment:
In the original code, there were two loops and thus `break` only applied to the inner. I think because of the override proxy mechanism, we should not have a break at all.

Reply:
Can you add a test where we have an unimplemented error for two different functions in the same file?

Reply:
(or better a function and a modifier)

Reply:
https://github.com/ethereum/solidity/pull/8659/files#diff-ae458dfbbab7ab6bd6ee10868650fac4
@chriseth Here is a new one I added.